### DISCUSSION TITLE
K8s cron job with bash scripts vs tool commands

### DISCUSSION THREAD

**Discussion by marrrcin** [Altinity/clickhouse-backup#846] [Q&A] [answered]
Hi,
thanks for creating this tool!

I have a difficulty understanding the documentation related to backup/restore on Kubernetes. In the base example of doing backup/restore the tool is used by invoking... the tool itself:
```bash
shard_number=$(clickhouse-client -q "SELECT getMacro('shard')")
clickhouse-backup create_remote shard${shard_number}-backup
clickhouse-backup delete local shard${shard_number}-backup
```
while a few pages below, there is an example of a k8s cron job that actually does the `INSERT`s into the database which seems counter intuitive and "manual" so to say (and uses `clickhouse-client`)

```bash
clickhouse-client --echo -mn -q "INSERT INTO system.backup_actions(command) VALUES('create ${SERVER}-${BACKUP_NAMES[$SERVER]}')" --host="$SERVER" --port="$CLICKHOUSE_PORT" --user="$BACKUP_USER" $BACKUP_PASSWORD;
```

Is this just a leftover in readme or is there some particular reasoning why the tool cannot be run in cron job? Could you provide some context on that? It would help a lot. Thanks!

**Comment by Slach** [comment, marked as answer]
In kubernetes we use `clickhouse-operator` which deploy `clickhouse-backup` as container inside the Pod where already present `clickhouse-server` and execute `clickhouse-backup server` in REST API mode, which create engine=url() tables inside `clickhouse-server`

`CronJob` executes outside Pods with `clickhouse-server + clickhouse-backup`, and execute the same command via SQL queries

So I would like to recommend use `clickhouse-backup` cli interface for stanalone installations

And clickhouse-backup REST API for containerized installations

**Reply by Slach** [reply]
clickhouse-backup can't be executed inside CronJob cause it require call kubernetes API and could be disabled for security reasons

**Reply by marrrcin** [reply]
Thanks for a quick response! This make sense now alongside the 48:12 of this video https://youtu.be/megsNh9Q-dw?si=7q1EFsKfTawY0Ro4&t=2892 :)
